File: 06_Navigation/06_NavigationDrawer/02_DrawerWithToolbar/src/main/java/com/codingnight/example/navigationdrawer/ThirdFragment.java
package com.codingnight.example.navigationdrawer;

import android.animation.ObjectAnimator;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import java.util.Random;

public class ThirdFragment extends Fragment {

    private ThirdViewModel mViewModel;
    private ImageView imageView;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.third_fragment, container, false);
        imageView = view.findViewById(R.id.imageView);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        mViewModel = new ViewModelProvider(requireActivity()).get(ThirdViewModel.class);
        imageView.setX(imageView.getX() + mViewModel.dX);
        // TODO: Use the ViewModel
        final ObjectAnimator objectAnimator = ObjectAnimator.ofFloat(imageView, "x", 0, 0);
        objectAnimator.setDuration(500);
        imageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!objectAnimator.isRunning()) {
                    float dx = new Random().nextBoolean() ? 100 : -100;
                    objectAnimator.setFloatValues(imageView.getX(), imageView.getX() + dx);
                    mViewModel.dX += dx;
                    objectAnimator.start();
                }
            }
        });
    }

}

File: 06_Navigation/06_NavigationDrawer/01_AppBarConfiguration/src/main/java/com/codingnight/example/navigationdrawer/ThirdFragment.kt
package com.codingnight.example.navigationdrawer

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.animation.ObjectAnimator
import android.view.View
import android.widget.ImageView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import java.util.*

class ThirdFragment : Fragment() {
    private var mViewModel: ThirdViewModel? = null
    private var imageView: ImageView? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.third_fragment, container, false)
        imageView = view.findViewById(R.id.imageView)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mViewModel = ViewModelProvider(requireActivity()).get(
            ThirdViewModel::class.java
        )
        imageView!!.x = imageView!!.x + mViewModel!!.dX
        // TODO: Use the ViewModel
        val objectAnimator = ObjectAnimator.ofFloat(imageView, "x", 0f, 0f)
        objectAnimator.duration = 500
        imageView!!.setOnClickListener {
            if (!objectAnimator.isRunning) {
                val dx: Float = if (Random().nextBoolean()) 100F else (-100).toFloat()
                objectAnimator.setFloatValues(imageView!!.x, imageView!!.x + dx)
                mViewModel!!.dX += dx
                objectAnimator.start()
            }
        }
    }
}